soes/esc_coe.c

```diff
@@ -80,9 +80,9 @@ int16_t SDO_findsubindex (int16_t nidx, uint8_t subindex)
  * @param[in] index   = value on index of object we want to locate
  * @return local array index if we succeed, -1 if we didn't find the index.
  */
-int32_t SDO_findobject (uint16_t index)
+int16_t SDO_findobject (uint16_t index)
```

Comment:
Theoretically there can be 65536 entries in the SDOobjects[] array. The function needs a signed result (possible -1) and  therefore the smallest type would be int32_t. The only change needed is promotion of int to int32_t at line 1863.

Reply:
fixed